### PR TITLE
Add distribution support to semantic-metrics FfwdhttpReporter

### PR DESCRIPTION
**Pull request by ao2017**
Add distribution support to ffwdhttpReporter.
PR depends on upcoming changes in ffwd-http-client.

### CODE REVIEW

File: ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java [hunk 1]
```diff
@@ -289,8 +299,12 @@ private BatchBuilder createBuilder(
         return new BatchBuilder(points, timestamp, key, id.getTags(), unit, metricType);
     }
 
+    private static Map<String, String> createResource() {
+        return new HashMap<>();
+    }
+
     private void reportGauge(
-        final BatchBuilder builder, @SuppressWarnings("rawtypes") Gauge value
+            final BatchBuilder builder, @SuppressWarnings("rawtypes") Gauge value
```
Comment (lmuhlha) [inline ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java:307]: just a nitpick, think there's extra space added

File: ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java [hunk 2]
```diff
@@ -83,63 +110,68 @@ public Double getValue() {
...
+            of("unit", "n/s", "stat", "5m", "metric_type", "deriving-meter"), RESOURCE, VALUE_0, TIME));
         expected.add(new Batch.Point("prefix.deriving-meter",
-            of("unit", "n/s", "stat", "1m", "metric_type", "deriving-meter"), 0, TIME));
+            of("unit", "n/s", "stat", "1m", "metric_type", "deriving-meter"), RESOURCE, VALUE_0, TIME));
         expected.add(new Batch.Point("prefix.histogram",
-            of("unit", "n", "stat", "max", "metric_type", "histogram"), 0, TIME));
+            of("unit", "n", "stat", "max", "metric_type", "histogram"), RESOURCE, VALUE_0, TIME));
         expected.add(new Batch.Point("prefix.histogram",
-            of("unit", "n", "stat", "min", "metric_type", "histogram"), 0, TIME));
+            of("unit", "n", "stat", "min", "metric_type", "histogram"), RESOURCE, VALUE_0, TIME));
```
Comment (lmuhlha) [inline ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java:128]: is batch point - key, tags, resource, value, and then timestamp? so you'd add a resource value but still need that 0 afterwards, right?
Similar to above: 
points.add(new Batch.Point(key, statsMap(stat), createResource(),
                    Value.DoubleValue.create(value), timestamp));

Comment (ao2017) [reply ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java:128]: Yes, now it is => key, tags, resource, value, and then timestamp. 
 VALUE_0 = Value.DoubleValue.create(0)
Resource is always empty. I hope that answer your question

File: ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java (hunk 1 above)
Comment (ao2017) [reply ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java:304]: For some reason, I get a checkstyle error when I removed the space :) . Was the first code submitted before this checkstyle ?

Comment (lmuhlha) [reply ffwd-http-reporter/src/main/java/com/spotify/metrics/ffwdhttp/FastForwardHttpReporter.java:304]: hm that's weird, leave it as is then. don't mind me :)

File: ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java (hunk 2 above)
Comment (lmuhlha) [reply ffwd-http-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardHttpReporterTest.java:128]: ah i was reading the , as _. this is what happens when i review in the morning lol.

**Review by lmuhlha — APPROVED** [review]

**Pull request merged**